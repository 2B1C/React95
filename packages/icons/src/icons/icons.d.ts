interface ICO extends ArrayBuffer {}

declare module '*.ico' {
  const value: ICO;
  export = value;
}
