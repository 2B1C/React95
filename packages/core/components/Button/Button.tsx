import Btn from '../shared-style/Btn';

const Button = Btn;

Button.defaultProps = {
  children: 'Ok',
};

export default Button;
