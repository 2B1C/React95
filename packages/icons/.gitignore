src/react/
src/index.ts
src/iconDemos.ts
png/
icons.css
