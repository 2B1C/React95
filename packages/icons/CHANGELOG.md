# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.2](https://github.com/React95/React95/compare/@react95/icons@0.4.1...@react95/icons@0.4.2) (2020-10-03)


### Bug Fixes

* **icons:** remove dictories and file fields from package.json and fix publishConfig ([2d788cc](https://github.com/React95/React95/commit/2d788cccaf41517e52cffc417731cba4f7e97087))





## [0.4.1](https://github.com/React95/React95/compare/@react95/icons@0.4.0...@react95/icons@0.4.1) (2020-10-03)


### Bug Fixes

* **icons:** add prepublish npm script ([cca8196](https://github.com/React95/React95/commit/cca8196129007bb5b33ba600cec5dd6bf13c678a))





# [0.4.0](https://github.com/React95/React95/compare/@react95/icons@0.3.0...@react95/icons@0.4.0) (2020-10-03)


### Features

* **icons:** add icons declaration file ([893b3ef](https://github.com/React95/React95/commit/893b3ef7f2fd639cdc1766907e4f62608bbfd8a3))
* **types:** move *.ico module declaration to root types folder ([bef505d](https://github.com/React95/React95/commit/bef505d89fcdfcbb1da7bad855f9eb03308191ba))





# [0.3.0](https://github.com/React95/React95/compare/@react95/icons@0.2.0...@react95/icons@0.3.0) (2020-06-14)


### Features

* **icons:** add new icons! :tada: ([feaf970](https://github.com/React95/React95/commit/feaf97094bdb7da37fae82d14871286e85e74ab4)), closes [#154](https://github.com/React95/React95/issues/154)





# [0.2.0](https://github.com/React95/React95/compare/@react95/icons@0.1.0...@react95/icons@0.2.0) (2020-05-15)


### Features

* **icons:** a lot of new icons! ([ceb0761](https://github.com/React95/React95/commit/ceb07614de289fba86e837dacf7399313e4a1460))





# 0.1.0 (2019-09-15)


### Features

* **Icon:** Add icons package ([94ec2af](https://github.com/React95/React95/commit/94ec2af)), closes [#124](https://github.com/React95/React95/issues/124)
