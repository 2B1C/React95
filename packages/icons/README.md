# `@react95/icons`

> TODO: description

## Usage

```
const icons = require('@react95/icons');

// TODO: DEMONSTRATE API
```
