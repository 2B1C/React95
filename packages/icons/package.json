{
  "name": "@react95/icons",
  "version": "1.0.3",
  "description": "React95 icons gallery",
  "keywords": [
    "React95",
    "icons",
    "windows95",
    "w95"
  ],
  "files": [
    "icons.css",
    "dist",
    "png"
  ],
  "author": "ggdaltoso <ggdaltoso@gmail.com>",
  "homepage": "https://react95.github.io/React95/",
  "license": "MIT",
  "main": "./src/index.ts",
  "types": "./dist/types/",
  "unpkg": "icons.css",
  "publishConfig": {
    "access": "public",
    "directory": "dist"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/React95/React95.git"
  },
  "bugs": {
    "url": "https://github.com/React95/React95/issues"
  },
  "scripts": {
    "precopy": "mkdir -p dist/src/png",
    "postinstall": "ts-node processIcons.ts",
    "build": "yarn build:cjs && yarn build:esm && yarn build:types",
    "build:types": "cross-env NODE_ENV=production tsc -p ./tsconfig.json",
    "build:esm": "cross-env NODE_ENV=esm yarn babel ./src --out-dir dist/esm --extensions .ts,.tsx",
    "build:cjs": "cross-env NODE_ENV=cjs yarn babel ./src --out-dir dist/cjs --extensions .ts,.tsx",
    "prepublish": "yarn build",
    "prepublishOnly": "node ../../scripts/prepublish.js"
  },
  "devDependencies": {
    "@types/rimraf": "^3.0.0",
    "rimraf": "^3.0.2",
    "ts-node": "^9.1.1",
    "typescript": "^4.2.3"
  }
}
