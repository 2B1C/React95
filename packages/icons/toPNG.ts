import fs from 'fs';
import ICO from 'icojs';

const ICONS_FOLDER = './src/icons';
const DEST_FOLDER = './src/png';

const files = fs.readdirSync(ICONS_FOLDER);

files.forEach(icon => {
  const buffer = fs.readFileSync(`${ICONS_FOLDER}/${icon}`);

  ICO.parse(buffer, 'image/png').then(images => {
    images.forEach(image => {
      const [iconName] = icon.split('.');

      let fileName = `${iconName}_${image.width}x${image.height}-${image.bpp}`;

      if (fs.existsSync(`${DEST_FOLDER}/${fileName}bit.png`)) {
        fileName = `${fileName}bit_(1).png`;
      } else {
        fileName = `${fileName}bit.png`;
      }

      const data = Buffer.from(image.buffer);
      const fileDestination = `${DEST_FOLDER}/${fileName}`;

      fs.writeFileSync(fileDestination, data);
      console.log(`icon ${fileDestination} created`);
    });
  });
});


  
