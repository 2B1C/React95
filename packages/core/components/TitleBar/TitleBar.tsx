import React from 'react';
import styled from '@xstyled/styled-components';
import { th } from '@xstyled/system';
import { BatExec } from '@react95/icons';

import Button from '../Button';

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const OptionItem = styled.li`
  display: flex;
  margin-left: 2;

  &:first-child {
    margin-right: 0;
  }
`;

const Option = styled(Button)`
  padding: 0;

  width: 17px;
  height: 14px;

  min-width: 0;
  font-weight: 600;
  font-size: 10;

  &:active {
    padding: 1 0 0 1;

    outline: none;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
      inset -1px -1px 0px 1px ${th('colors.borderDark')};
  }
`;

Option.displayName = 'Option';

interface TitleBarBackgroundProps extends React.HTMLAttributes<HTMLDivElement> {
  isActive?: boolean;
  icon?: React.ReactElement;
  title?: string;
}

const TitleBarBackground = styled.div<{ isActive?: boolean }>`
  height: 18px;
  margin-bottom: 2;

  padding: 2 2 0;

  display: flex;

  img {
    width: 15px;
    height: 13px;
    margin-right: 4px;
  }
  background: ${({ isActive, theme }) =>
    isActive
      ? theme.colors.headerBackground
      : theme.colors.headerNotActiveBackground};
  color: ${({ isActive, theme }) =>
    isActive ? theme.colors.headerText : theme.colors.headerNotActiveText};
`;

const Title = styled.div`
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
`;

interface TitleBarOptions {
  Option: typeof Option;
  OptionItem: typeof OptionItem;
  OptionsBox: typeof OptionsBox;
}

export interface ITitleBarProps
  extends TitleBarBackgroundProps,
    TitleBarOptions {}

const TitleBarRenderer: React.ForwardRefRenderFunction<
  HTMLDivElement,
  TitleBarBackgroundProps
> = (
  {
    children,
    title = 'UNKNOWN.EXE',
    icon = <BatExec variant="16x16_4" />,
    isActive = true,
    className,
    ...props
  },
  ref: React.Ref<HTMLDivElement>,
) => (
  <TitleBarBackground isActive={isActive} {...props} ref={ref}>
    {icon && icon}
    <Title className={className}>{title}</Title>

    {children}
  </TitleBarBackground>
);

const TitleBar = Object.assign(
  React.forwardRef<HTMLDivElement, TitleBarBackgroundProps>(TitleBarRenderer),
  {
    Option,
    OptionItem,
    OptionsBox,
  },
) as ITitleBarProps & typeof TitleBarRenderer;

export default TitleBar;
