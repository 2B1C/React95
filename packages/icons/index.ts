import icons from './src';

export default icons;
