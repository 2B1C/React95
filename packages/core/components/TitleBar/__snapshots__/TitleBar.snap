// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar /> Snapshot should match snapshot 1`] = `
.c0 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #000e7a;
  color: #fefefe;
}

.c0 img {
  width: 15px;
  height: 13px;
  margin-right: 4px;
}

.c2 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #7f787f;
  color: #c6c6c6;
}

.c2 img {
  width: 15px;
  height: 13px;
  margin-right: 4px;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

<div>
  <div
    class="c0"
  >
    <img
      height="32"
      src="test-file-stub"
      width="32"
    />
    <div
      class="c1"
    >
      test.exe
    </div>
  </div>
  <div
    class="c2"
  >
    <img
      height="32"
      src="test-file-stub"
      width="32"
    />
    <div
      class="c1"
    >
      test.exe
    </div>
  </div>
  <div
    class="c0"
  >
    <img
      height="16"
      src="test-file-stub"
      width="16"
    />
    <div
      class="c1"
    >
      UNKNOWN.EXE
    </div>
  </div>
  <div
    class="c2"
  >
    <img
      height="16"
      src="test-file-stub"
      width="16"
    />
    <div
      class="c1"
    >
      UNKNOWN.EXE
    </div>
  </div>
</div>
`;

exports[`<TitleBar /> Snapshot should match snapshot with all Options 1`] = `
.c4 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c4:disabled {
  color: #868a8e;
}

.c4:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c4:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c2 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 2px;
}

.c3:first-child {
  margin-right: 0;
}

.c5 {
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-weight: 600;
  font-size: 10px;
}

.c5:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c5:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
}

.c0 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #000e7a;
  color: #fefefe;
}

.c0 img {
  width: 15px;
  height: 13px;
  margin-right: 4px;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

<div>
  <div
    class="c0"
  >
    <img
      height="32"
      src="test-file-stub"
      width="32"
    />
    <div
      class="c1"
    >
      test.exe
    </div>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <button
          class="c4 c5"
        >
          ?
        </button>
      </li>
      <li
        class="c3"
      >
        <button
          class="c4 c5"
        >
          x
        </button>
      </li>
    </ul>
  </div>
</div>
`;
